day46/problem.go

```diff
@@ -0,0 +1,24 @@
+package day46
+
+// IsPalindrome answers this question is linear time O(N) and O(1) space.
+func IsPalindrome(letters []rune) bool {
+	for i := 0; i < len(letters)/2; i++ {
+		if letters[i] != letters[len(letters)-1-i] {
+			return false
+		}
+	}
+	return true
+}
+
+// LongestPalindromicSubstringBrute runs in O(N^2) time and O(1) space.
```

Comment:
It’s actually O(N^3) because the IsPalindrome check is linear nested inside of for loops.